lib/nerves_ssh.ex

```diff
@@ -26,18 +26,24 @@ defmodule NervesSSH do
     defstruct opts: [], sshd: nil, sshd_ref: nil
   end
 
+  defp via_name(name), do: {:via, Registry, {NervesSSH.Registry, name}}
+
   @doc false
-  @spec start_link(Options.t()) :: GenServer.on_start()
+  @spec start_link(Options.t() | {atom(), Options.t()}) :: GenServer.on_start()
+  def start_link({name, %Options{} = opts}) do
```

Comment:
I think since we now need `:name` as part of the options, we should remove this start_link function header and only rely on the one below expecting the options struct.